classes/DepositPointDAO.inc.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/**
+ * @file plugins/generic/sword/classes/DepositPointDAO.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class DepositPointDAO
+ * @ingroup plugins_generic_sword_classes
+ *
+ * @brief Operations for retrieving and modifying DepositPoint objects.
+ */
+
+import('lib.pkp.classes.db.DAO');
+
+class DepositPointDAO extends DAO {
+	/** @var SwordPlugin reference to SWORD plugin */
+	protected $_plugin = null;
+
+	/**
+	 * Constructor
+	 * @param $parentPlugin SwordPlugin
+	 */
+	public function __construct(SwordPlugin $parentPlugin) {
+		$this->_plugin = $parentPlugin;
+		parent::__construct();
+	}
+
+	/**
+	 * Instantiate a new data object.
+	 * @return DepositPoint
+	 */
+	public function newDataObject() {
+		$this->_plugin->import('classes.DepositPoint');
+		return new DepositPoint();
+	}
+
+	/**
+	 * Retrieve a deposit point by ID.
+	 * @param $depositPointId int
+	 * @param $contextId int
+	 * @return DepositPoint
+	 */
+	public function getById($depositPointId, $contextId = null) {
+		$params = array((int) $depositPointId);
+		if ($contextId) $params[] = (int) $contextId;
+		
+		$result = $this->retrieve(
+			'SELECT * FROM deposit_points WHERE deposit_point_id = ? ' . ($contextId?' AND context_id = ?':''),
+			$params
+		);
+		
+		$returner = null;
+		if ($result->RecordCount() != 0) {
+			$returner = $this->_fromRow($result->GetRowAssoc(false));
+		}
+		$result->Close();
+		return $returner;
+	}
+
+	/**
+	 * Internal function to return DepositPoint object from a row.
+	 * @param $row array
+	 * @return DepositPoint
+	 */
+	public function _fromRow($row) {
```

Comment:
This should be `protected` (per the `_` prefix).

Reply:
done.

Reply:
making this method protected results into the following error:
`PHP Fatal error:  Call to protected method DepositPointDAO::_fromRow() from context 'DAOResultFactory' in /www/pkp/sword/lib/pkp/classes/db/DAOResultFactory.inc.php on line 104`